### PR TITLE
Implement the `favourited_by` and `reblogged_by` API calls.

### PR DESCRIPTION
**Pull request by tchoutri**
@milmazz Tell me if I messed up something about the architecture ;) I tried to stick to what you did until now.

### CODE REVIEW

**Comment by milmazz** [comment]
Hello, @tchoutri! This is your first Pull Request that will be reviewed by [Ebert](https://ebertapp.io/), an automatic Code Review service. It will leave comments on this diff with potential issues and style violations found in the code as you push new commits. You can also see all the issues found on this Pull Request on its [review page](https://ebertapp.io/github/milmazz/hunter/pulls/2). Please check our [documentation](https://ebertapp.io/docs) for more information.

File: lib/hunter/status.ex [hunk 1]
```diff
@@ -185,6 +200,21 @@ defmodule Hunter.Status do
   end
 
   @doc """
+  Fetch the list of users who favourited the status
+
```
Comment (milmazz) [inline lib/hunter/status.ex]: There should be no trailing white-space at the end of a line.

File: lib/hunter/api.ex [hunk 2]
```diff
@@ -295,6 +295,17 @@ defmodule Hunter.Api do
   """
   @callback unreblog(conn :: Hunter.Client.t, id :: non_neg_integer) :: Hunter.Status.t
 
+
```
Comment (milmazz) [inline lib/hunter/api.ex]: There should be no more than 1 consecutive blank lines.

**Comment by milmazz** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 2 possible new issues (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/milmazz/hunter/pulls/2.

**Comment by milmazz** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 1 possible new issue (including those that may have been commented here).
* 1 fixed issue! 🎉

You can see more details about this review at https://ebertapp.io/github/milmazz/hunter/pulls/2.

File: lib/hunter.ex [hunk 3]
```diff
@@ -333,6 +333,17 @@ defmodule Hunter do
   defdelegate unreblog(conn, id), to: Hunter.Status
 
   @doc """
+  Fetch the list of users who reblogged the status.
+
+  ## Parameters
+
+    * `conn` - connection credentials
+    * `id` - status identifier
+  """
```
Comment (milmazz) [inline lib/hunter.ex]: Please add a newline after the end of the parameters list

File: lib/hunter.ex [hunk 4]
```diff
@@ -333,6 +333,17 @@ defmodule Hunter do
...
+  @spec reblogged_by(Hunter.Client.t, non_neg_integer) :: Hunter.Status.t
+  defdelegate reblogged_by(conn, id), to: Hunter.Status
```
Comment (milmazz) [inline lib/hunter.ex]: I think this endpoint should return a list of accounts: `[Hunter.Account.t]` seems more appropriate according to your description.

File: lib/hunter/api.ex [hunk 5]
```diff
@@ -295,6 +295,17 @@ defmodule Hunter.Api do
...
+  @doc """
+  Fetch the list of users who reblogged the status.
+
+  ## Parameters
+
+    * `conn` - connection credentials
+    * `id` - status identifier
+  """
```
Comment (milmazz) [inline lib/hunter/api.ex]: Please add a newline after the end of the parameters list

File: lib/hunter/api.ex [hunk 6]
```diff
@@ -328,6 +339,16 @@ defmodule Hunter.Api do
   @callback favourites(conn :: Hunter.Client.t) :: [Hunter.Status.t]
 
   @doc """
+  Fetch the list of users who favourited the status.
+
+  ## Parameters
+
+    * `conn` - connection credentials
+    * `id` - status identifier
+  """
```
Comment (milmazz) [inline lib/hunter/api.ex]: Please add a newline after the end of the parameters list

File: lib/hunter/api/http_client.ex [hunk 7]
```diff
@@ -53,6 +53,7 @@ defmodule Hunter.Api.HTTPClient do
   end
 
   def create_app(%Hunter.Client{base_url: base_url} = conn, name, redirect_uri, scopes, website) do
+    IO.puts redirect_uri
```
Comment (milmazz) [inline lib/hunter/api/http_client.ex]: Please remove this line

File: lib/hunter/status.ex [hunk 8]
```diff
@@ -138,12 +140,25 @@ defmodule Hunter.Status do
...
   end
 
   @doc """
+  Fetch the list of users who reblogged the status.
+
+  ## Parameters
+
+    * `conn` - connection credentials
+    * `id` - status identifier
+  """
```
Comment (milmazz) [inline lib/hunter/status.ex]: Please add a newline after the end of the parameters list

File: lib/hunter/status.ex [hunk 9]
```diff
@@ -82,7 +84,7 @@ defmodule Hunter.Status do
     * `media_ids` - [Array<Integer>]
 
   """
-  @spec create_status(Hunter.Client.t, String.t, non_neg_integer, [non_neg_integer]) :: Hunter.Status.t
+  @spec create_status(Hunter.Client.t, String.t, status_id, [status_id]) :: Hunter.Status.t
```
Comment (milmazz) [inline lib/hunter/status.ex]: If you want to be explicit, which is fine, why are you assuming that `media_ids` it's the same type as `status_id`?

File: test/support/in_memory.ex [hunk 10]
```diff
@@ -100,6 +100,10 @@ defmodule Hunter.Api.InMemory do
     %Hunter.Status{}
   end
 
+  def reblogged_by(_, _) do
```
Comment (milmazz) [inline test/support/in_memory.ex]: Could you please add a fixture, then include this in the list at the beginning of this file?

**Comment by tchoutri** [comment]
This is just a preliminary push, I need to have an overview of the situation with more than just `git diff`.

**Comment by milmazz** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 2 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/milmazz/hunter/pulls/2.

**Comment by milmazz** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 5 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/milmazz/hunter/pulls/2.

**Comment by milmazz** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 4 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/milmazz/hunter/pulls/2.

File: lib/hunter/status.ex [hunk 11]
```diff
@@ -138,12 +140,26 @@ defmodule Hunter.Status do
...
   @doc """
+  Fetch the list of users who reblogged the status.
+
+  ## Parameters
+
+    * `conn` - connection credentials
+    * `id` - status identifier
+
+  """
+  @spec reblogged_by(Hunter.Client.t, non_neg_integer) :: [Hunter.Account.t]
```
Comment (milmazz) [inline lib/hunter/status.ex]: s/non_neg_status/status_id/

Comment (tchoutri) [inline lib/hunter/status.ex]: Alright :)
I ping'd you the Elixir Slack, btw. So you can reach me if you need.

**Comment by milmazz** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 4 fixed issues! 🎉

But beware that this branch is **[1 commit behind the `milmazz:master` branch](https://github.com/milmazz/hunter/compare/tchoutri:favd-boostd-by...milmazz:master)**, and a review of an up to date branch would produce more accurate results.

You can see more details about this review at https://ebertapp.io/github/milmazz/hunter/pulls/2.

**Comment by milmazz** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 4 fixed issues! 🎉

You can see more details about this review at https://ebertapp.io/github/milmazz/hunter/pulls/2.

**Comment by milmazz** [comment]
@tchoutri Thanks a lot! :1st_place_medal:

**Pull request merged**